lib/periodicChecks.js

```diff
@@ -0,0 +1,49 @@
+const assignUnassignedReviewers = async (pullRequest, context) => {
+  const assignees = pullRequest.assignees.map(user => user.login);
+  const reviewers = pullRequest.requested_reviewers.map(user => user.login);
+
+  // If the PR author is assigned, then don't assign other reviewers.
+  if (assignees.includes(pullRequest.user.login)) {
+    return;
+  }
+
+  let botComment = 'Hey ';
+  const prParams = context.repo(
+    {
+      number: pullRequest.number,
+    }
+  );
+
+  const unassignedReviewers = reviewers.filter(reviewer => {
+    if (!assignees.includes(reviewer)) {
+      botComment += `@${reviewer}, `;
+      return true;
+    }
+  });
+  if (unassignedReviewers.length > 0) {
+    await context.github.issues.addAssigneesToIssue(
+      { ...prParams, assignees: unassignedReviewers });
+
+    // Ping newly assigned reviewers.
+    botComment += 'this PR is waiting for your review, can you PTAL? Thanks!';
+    await context.github.issues.createComment(
+      { ...prParams, body: botComment });
+  }
+};
+
+module.exports.assignReviewers = async (context) => {
+  // By default, this returns only open PRs.
+  const pullRequestsPromiseObj = await context.github.pullRequests.getAll(
+    context.repo({ per_page: 60 })
+  );
+
+  const arrayOfOpenPullRequests = pullRequestsPromiseObj.data;
+
+  arrayOfOpenPullRequests.forEach(async (pullRequest) => {
+    const pullRequestDetailsPromise = await context.github.pullRequests.get(
+      context.repo({ number: pullRequest.number })
+    );
+    const pullRequestDetails = pullRequestDetailsPromise.data;
+    await assignUnassignedReviewers(pullRequestDetails, context);
```

Comment:
ditto here can we assign reviewers to a list of PRs at once? (not sure, can you please check whether that's possible?)

Reply:
The reviewers may not be the same so I don't think this will be feasible

Reply:
Yeah, we can generate a comment message for all the pull request one at a time and make one call to publish all the comments (if that's possible?)?

Reply:
I think the comment should be specific (including the reviewers). Also, I confirmed that [creating a comment](https://octokit.github.io/rest.js/v17#issues-create-comment) requires the issue or PR number, and not an array of numbers so it wouldn't be possible to publish it across multiple PRs.

Reply:
I mean is there any API to publish multiple comments at once:
```
comments: [{
  number: ...,
  body: botComment1
}, {
  number: ...,
  body: botComment1
},..]
```

This way we won't have to make multiple https calls.

Also, I to had this concern (image below): 
![image](https://user-images.githubusercontent.com/16653571/79149202-6ba1c200-7de4-11ea-80d3-bb6d5a4864a8.png)
Considering this can happen, should we log exception for [403 returns](https://developer.github.com/v3/#abuse-rate-limits)? Also, maybe follow [these guidelines](https://developer.github.com/v3/guides/best-practices-for-integrators/#dealing-with-abuse-rate-limits)?

[In general, how do we log issues (400* errors) raised while calling API?]

Reply:
I think we should handle 400 or 500 errors otherwise we will be making unwanted/unauthorized calls multiple time.
`Intentionally ignoring repeated validation errors may result in the suspension of your app for abuse.` ([Source](https://developer.github.com/v3/guides/best-practices-for-integrators/#dealing-with-api-errors))

I'm not sure how probot handles these errors, so I just wanted to check we are aware of this.

Reply:
Ohh
I'm sorry I missed this.
> I mean is there any API to publish multiple comments at once:
No, the API does not support creating multiple comments at [once ](https://octokit.github.io/rest.js/v17#issues-create-comment)

> I'm not sure how probot handles these errors, so I just wanted to check we are aware of this.

We are not making the requests directly since we are using the [github octokit](http://github.com/octokit/rest.js/) library. And the library specifies what contents should be sent so I don't think we will fall into this issue. 
I dug deeper, and probot provides a SENTRY_DSN env variable for error logging to sentry.io https://probot.github.io/docs/configuration/

Reply:
> And the library specifies what contents should be sent so I don't think we will fall into this issue.

But the library itself says that don't make many comments quickly and redirects to a URL which has these guidelines ([ref](https://octokit.github.io/rest.js/v17#issues-create-comment))

> I dug deeper, and probot provides a SENTRY_DSN env variable for error logging to sentry.io https://probot.github.io/docs/configuration/

But for that we need to have a Sentry account, right?  Can you please check how to handle these http errors on probot API. Like what will happen if the add comment API doesn't work correctly.